osu.Game/Rulesets/RulesetStore.cs

```diff
@@ -17,16 +18,20 @@ public class RulesetStore : DatabaseBackedStore, IDisposable
 
         private readonly Dictionary<Assembly, Type> loadedAssemblies = new Dictionary<Assembly, Type>();
 
-        public RulesetStore(IDatabaseContextFactory factory)
+        private readonly Storage rulesetStorage;
+
+        public RulesetStore(IDatabaseContextFactory factory, Storage storage = null)
             : base(factory)
         {
+            rulesetStorage = storage?.GetStorageForDirectory("rulesets");
+
             // On android in release configuration assemblies are loaded from the apk directly into memory.
             // We cannot read assemblies from cwd, so should check loaded assemblies instead.
             loadFromAppDomain();
             loadFromDisk();
+            AppDomain.CurrentDomain.AssemblyResolve += resolveRulesetDependencyAssembly;
```

Comment:
As I commented before, is there a reason why this is done in the specific place it's being done here? Rather than at the bottom where it was initially?

Reply:
The event handler contains code for resolving dependency on the game assembly for Rulesets located outside the base game directory. 

It needs to be attached  to the lookup event before the actual call to `loadUserRulesets()` else Rulesets located out the base game directory will fail to load as unable to locate the game assemblies.

Reply:
Gotcha, thanks.

Reply:
Maybe could have a comment here why this needs to be before the user ruleset load?